packages/flutter_tools/lib/src/cache.dart

```diff
@@ -560,13 +564,16 @@ class FlutterEngine extends CachedArtifact {
     return result;
   }
 
-
   void _makeFilesExecutable(Directory dir) {
-    for (FileSystemEntity entity in dir.listSync()) {
+    os.chmod(dir, 'a+r,a+x');
```

Comment:
Having a test that the permissions are set correctly alongside the artifact tests would be good:

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/artifacts_test.dart

Reply:
Done